src/connection.js

```diff
@@ -585,8 +594,12 @@ class Connection extends EventEmitter {
 
   requestTimeout() {
     this.requestTimer = undefined;
-    this.messageIo.sendMessage(TYPE.ATTENTION);
-    return this.transitionTo(this.STATE.SENT_ATTENTION);
+    this.cancel();
```

Comment:
This will set canceled=true which in turn will cause the error message to be Canceled instead of Timeout. You should have three different error messages for the three different cases - user initiated cancel(), request timer timeout and cancel timer timeout (i.e. timeout waiting for Attention acknowledgement)